source/server/getting-started/configuration/sponge-conf.rst

```diff
@@ -13,464 +13,1427 @@ Below is a list of available config settings and their comments inside the globa
-    a block type will be performed.
-
-block-tracking
-==============
+.. _ConfigType_GlobalConfig:
 
-block-blacklist
-    Block IDs that will be blacklisted for player block placement tracking.
-enabled
-    If 'true', adds player tracking support for block positions.
-    Note: This should only be disabled if you do not care who caused a block to change.
+GlobalConfig
+============
 
-bungeecord
-==========
+| The main configuration for Sponge: ``global.conf``
+|
+
+* **broken-mods**
+
+  | Stopgap measures for dealing with broken mods
+  | **Type:** :ref:`BrokenMod<ConfigType_BrokenMod>`
+  |
+
+* **bungeecord**
+
+  | **Type:** :ref:`BungeeCord<ConfigType_BungeeCord>`
+  |
+
+* **cause-tracker**
+
+  | **Type:** :ref:`PhaseTracker<ConfigType_PhaseTracker>`
+  |
+
+* **commands**
+
+  | **Type:** :ref:`Commands<ConfigType_Commands>`
+  |
+
+* **debug**
+
+  | **Type:** :ref:`Debug<ConfigType_Debug>`
+  |
+
+* **entity**
+
+  | **Type:** :ref:`Entity<ConfigType_Entity>`
+  |
+
+* **entity-activation-range**
+
+  | **Type:** :ref:`EntityActivationRange<ConfigType_EntityActivationRange>`
+  |
+
+* **entity-collisions**
+
+  | **Type:** :ref:`EntityCollision<ConfigType_EntityCollision>`
+  |
+
+* **exploits**
+
+  | **Type:** :ref:`Exploit<ConfigType_Exploit>`
+  |
+
+* **general**
+
+  | **Type:** :ref:`GlobalGeneral<ConfigType_GlobalGeneral>`
+  |
+
+* **ip-sets**
+
+  | **Type:** ``Map<String, List<IpSet>>``
+  |
+
+* **logging**
 
-ip-forwarding
-    If 'true', allows BungeeCord to forward IP address, UUID, and Game Profile to this server.
+  | **Type:** :ref:`Logging<ConfigType_Logging>`
+  |
 
-cause-tracker
+* **metrics**
+
+  | **Type:** :ref:`Metrics<ConfigType_Metrics>`
+  |
+
+* **modules**
+
+  | **Type:** :ref:`Module<ConfigType_Module>`
+  |
+
+* **movement-checks**
+
+  | **Type:** :ref:`MovementChecks<ConfigType_MovementChecks>`
+  |
+
+* **optimizations**
+
+  | **Type:** :ref:`Optimization<ConfigType_Optimization>`
+  |
+
+* **player-block-tracker**
+
+  | **Type:** :ref:`PlayerBlockTracker<ConfigType_PlayerBlockTracker>`
+  |
+
+* **spawner**
+
+  | Used to control spawn limits around players. 
+  | **Note**: The radius uses the lower value of mob spawn range and server's view distance.
+  | **Type:** :ref:`Spawner<ConfigType_Spawner>`
+  |
+
+* **sql**
+
+  | Configuration options related to the Sql service, including connection aliases etc
+  | **Type:** :ref:`Sql<ConfigType_Sql>`
+  |
+
+* **teleport-helper**
+
+  | Blocks to blacklist for safe teleportation.
+  | **Type:** :ref:`TeleportHelper<ConfigType_TeleportHelper>`
+  |
+
+* **tileentity-activation**
+
+  | **Type:** :ref:`TileEntityActivation<ConfigType_TileEntityActivation>`
+  |
+
+* **timings**
+
+  | **Type:** :ref:`Timings<ConfigType_Timings>`
+  |
+
+* **world**
+
+  | **Type:** :ref:`GlobalWorld<ConfigType_GlobalWorld>`
+  |
+
+.. _ConfigType_AsyncLighting:
+
+AsyncLighting
```

Comment:
Titles like this aren't good. This is exposing internal class names and may not match the header name. These headers really should be the config header names, as that's what people see.

I do understand how horrible this all is though, I don't think this is a deal breaker but I'm certainly edgy about it.

Reply:
I will check whether all the types are referenced only from one place/referenced with a unique name.

How should we handle the kebab-case names for the titles?

Reply:
Ideally these would be substituted for their matching config entry. That's the optimal case, but if we have to compromise to save twiddling the config file every time it updates then it's still OK, as long as users can find the entries they need. We want to make this easy for *us*, so we can allow the config entries to be translated on Crowdin.